src/bytes.rs

```diff
@@ -16,16 +16,38 @@ use safer_ffi_proc_macros::derive_ReprC;
 #[repr(C)]
 #[cfg_attr(feature = "stabby", stabby::stabby)]
 pub struct Bytes<'a> {
-    start: core::ptr::NonNull<u8>,
+    start: *const u8,
     len: usize,
     data: *const (),
     capacity: usize,
     vtable: &'a BytesVt,
 }
-
+#[cfg(not(feature = "stabby"))]
+unsafe impl<'a> crate::layout::__HasNiche__ for Bytes<'a> {
+    fn is_niche(it: &'_ <Self as crate::ඞ::ReprC>::CLayout) -> bool {
+        !it.vtable.is_null()
+    }
+}
 #[cfg(feature = "stabby")]
+unsafe impl<'a> crate::layout::__HasNiche__ for Bytes<'a>
+where
+    Bytes<'a>: stabby::IStable<HasExactlyOneNiche = stabby::abi::B1>,
```

Comment:
Optional/debatable nit, feel free to disregard this one:

```suggestion
    Self: stabby::IStable<HasExactlyOneNiche = stabby::abi::B1>,
```